src/ucp/core/ucp_context.c

```diff
@@ -1548,6 +1549,9 @@ ucs_status_t ucp_init_version(unsigned api_major_version, unsigned api_minor_ver
         ucp_config_release(dfl_config);
     }
 
+    ucs_vfs_obj_add_dir(NULL, context, "ucp/context/%s-%p", context->name,
+                        context);
```

Comment:
i think it can be just a name:
1. default name (if not provided) would be "%p"
2. ucs_vfs_obj_add_dir needs to fail if the name already exists

Reply:
do not you think it would be easier to match the context info with logs if its name contains pointer? Or maybe we can have a separate object showing its pointer then?

Reply:
better add it as separate object
BTW @rakhmets we also need to add API for worker and ep name (same as context)

Reply:
We can use the same logic for logs, e.g.:

```
if ((params->field_mask & UCP_PARAM_FIELD_NAME) && (params->name != NULL)) {
    ucs_snprintf_zero(context->name, UCP_CONTEXT_NAME_MAX, "%s", params->name);
} else {
    ucs_snprintf_zero(context->name, UCP_CONTEXT_NAME_MAX, "%p", context);
}
```

```
ucs_vfs_obj_add_dir(NULL, context, "ucp/context/%s", context->name);

ucs_debug("created ucp context %s [%d mds %d tls] features 0x%" PRIx64
          " tl bitmap " UCT_TL_BITMAP_FMT, context->name, context->num_mds, context->num_tls,
          context->config.features, UCT_TL_BITMAP_ARG(&context->tl_bitmap));
```

Reply:
+1, but in ucs_debug pls also add the pointer `%p` somewhere

Reply:
in this case have to change all existing logs

Reply:
> BTW @rakhmets we also need to add API for worker and ep name (same as context)

Why? Would not be context name enough? Imo, worker and ep names would complicate API/usage

Reply:
> in this case have to change all existing logs

What are the other places where context pointer value is printed? I only look through ucp_context.c.
Context name is already printed during initialization. This was added along with introducing new parameter field.

If there are only a couple of places where we print pointer value without context name then I can update them.

> +1, but in ucs_debug pls also add the pointer %p somewhere

However, if we need to print pointer value anyway and everywhere, then the initial naming of VFS object will be better, IMO.
`ucs_vfs_obj_add_dir(NULL, context, "ucp/context/%s-%p", context->name, context);`
in case of easy match logs with VFS tree is required.

Reply:
> What are the other places where context pointer value is printed? I only look through ucp_context.c.
> Context name is already printed during initialization. This was added along with introducing new parameter field.
> 
> If there are only a couple of places where we print pointer value without context name then I can update them.

Not sure how many existing logs need to be updated. If it is small enough, then ok. 

> 
> However, if we need to print pointer value anyway and everywhere, then the initial naming of VFS object will be better, IMO.
> `ucs_vfs_obj_add_dir(NULL, context, "ucp/context/%s-%p", context->name, context);`
> in case of easy match logs with VFS tree is required.

Maybe context can be an exception here as a root object, but for other objects I'd vote for pointers as a name

Reply:
just add name and pointer to one debug message of context create.
re worker, ep - I think it would be useful, for example can see endpoints by rank number, worker by which thread created them, etc.